neo4j/example/src/main/java/example/springdata/neo4j/ActorRepository.java

```diff
@@ -15,11 +15,15 @@
  */
 package example.springdata.neo4j;
 
+import java.util.UUID;
+
 import org.springframework.data.neo4j.repository.GraphRepository;
+import org.springframework.data.neo4j.repository.Neo4jRepository;
 
 /**
  * {@link GraphRepository} for {@link Actor}s.
- * 
+ *
  * @author Luanne Misquitta
+ * @author Mark Angrish
  */
-public interface ActorRepository extends GraphRepository<Actor> {}
+public interface ActorRepository extends Neo4jRepository<Actor, UUID> {}
```

Comment:
1. No it doesn't. In fact until you mentioned it I actually thought it was normal to extend the store specific interface. I'll change this.
1. Argh.. This has been a problem for a while now. The OGM and SDN 3.x both expose a neo4j database implementation called `ID` which currently needs to be annotated with `@GraphId`. Unlike other DB's it's *NOT* guaranteed to be unique across database restarts ([read more here](http://blog.armbruster-it.de/2013/08/assigning-uuids-to-neo4j-nodes-and-relationships/) - it's still relevant). On top of this developers have asked to have a way of uniquely identifying their domain objects, either naturally or with the Neo4j recommended approach of using `UUID`s.  Furthermore the implications of this result in things like: https://jira.spring.io/browse/DATAGRAPH-918 and https://github.com/spring-projects/spring-data-neo4j/issues/366. Ideally this implementation detail will be hidden in future. Personally I'm hoping when we can remove default constructors we can also remove the need for exposing the internal ID. What I will do for now is add some javadoc here to make this clear for those who haven't used Neo4j before.